src/split_train_test.py

```diff
@@ -0,0 +1,73 @@
+import os
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from deepchecks.tabular import Dataset
+from deepchecks.tabular.checks import DatasetsSizeComparison, TrainTestSamplesMix, MultivariateDrift, LabelDrift, FeatureDrift
+
+def split_train_test_data(cleaned_data, train_data_size, stratify_by=None):
+    """Split train test data using cleaned data"""
+
+    # Ensure cleaned_data is a dataframe
+    if not isinstance(cleaned_data, pd.DataFrame):
+        raise TypeError("The cleaned_data must be a pandas DataFrame.")
+
+    # Ensure train_data_size is a valid proportion
+    if not (0 < train_data_size < 1):
+        raise ValueError("train_data_size must be a float between 0 and 1.")
+
+    # Handle stratify_by
+    if stratify_by:
+        if stratify_by not in cleaned_data.columns:
+            raise KeyError(f"The column '{stratify_by}' does not exist in the cleaned_data.")
+    
+    # create the split
+    data_train, data_test = train_test_split(
+        cleaned_data, train_size=train_data_size, stratify=stratify_by #cleaned_data["diagnosis"]
+    )
+    
+    return data_train, data_test
+
+def validate_split_data(data_train, data_test):
+    """Test split datasets using Deepchecks"""
+
+    # test train test split dataframes
+    # prepare the train test datasets to Deepchecks Dataset format
+    data_train = Dataset(data_train,features=data_train.columns[1:],label=data_train.columns[0])
+    data_test = Dataset(data_test,features=data_test.columns[1:],label=data_test.columns[0])
```

Comment:
Please change from numerical subsetting to using column names/labels. The function can be made flexible to take in column names/labels. Numerical subsetting is very brittle and error-prone.